### DISCUSSION TITLE
Does DAFoam work with laminar cases?

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#633] [General]
Hi,

I want to run a case in laminar using DAFoam. 

My problem setup would be - I have a reference data for a cylinder case at Re 150 (for example). This was computed in OpenFOAM without a turbulence model. So I have the time averaged mean velocity on an unstructured grid. 

I want to use this as "UData" but run an optimization in DAFoam to optimize fvSource but without a turbulence model. I see that UData is available to me only if I use a turbulence model. So my two questions are as follows:

1. Can laminar cases be run using DAFoam?
2. How do i include UData if I am not calling any turbulence model?
Please advice, thank you very much!

**Comment by friedenhe** [comment]
Yes. Use our latest FIML interface. https://github.com/DAFoam/tutorials/blob/main/Ramp/steady/train/runScript_FI.py. It uses a new interface for the FIML objective, i.e., variance. So it reads UData directly from the objective function, instead of the turbulence model. This should work for laminar flow.

**Comment by uttamcadambi07** [comment]
Sounds good, thanks Dr. He

I am faced with a new problem.

I want to just do a ```runPrimal``` to begin with for a laminar case.

I have tried all permutations in the ```turbulenceProperties``` dictionary

```
simulationType laminar;

RAS 
{ 
    RASModel             laminar;
    turbulence           off;
    printCoeffs          off;
} 

```

being one of them,

I am always getting an error that goes like this

```
[0] 
[0] 
[0] --> FOAM FATAL ERROR: 
[0] Unknown DATurbulenceModel type laminar

Valid DATurbulenceModel types:

5
(
SpalartAllmaras
SpalartAllmarasDataAssimilation
SpalartAllmarasFieldInversion
SpalartAllmarasFieldInversionT
kOmegaSST
)

```

or if I do something like this
```
simulationType laminar;

/*
RAS 
{ 
    RASModel             laminar;
    turbulence           off;
    printCoeffs          off;
} 
*/

```

I get this error

```
Creating source term. 
Selecting incompressible transport model Newtonian
Reading field alphat

Selecting turbulence model type laminar
Selecting laminar stress model Stokes
No MRF models present

[0] 
[0] 
[0] --> FOAM FATAL IO ERROR: 
[0] Entry 'RAS' not found in dictionary "/home/uttamcadambi/cylinder_assimilation/constant/turbulenceProperties"
[0] 
[0] file: /home/uttamcadambi/cylinder_assimilation/constant/turbulenceProperties
[0] 
[0]     From function Foam::dictionary& Foam::dictionary::subDict(const Foam::word&)
[0]     in file db/dictionary/dictionary.C at line 493.

```

so what is the correct way to do this?

**Comment by friedenhe** [comment]
Use this setup for laminar flow. https://github.com/DAFoam/reg_test_files/blob/main/UBendDuct/constant/turbulenceProperties.dummy Also, it seems that you commented out many turbulence models in Make/files. Remember to keep the `dummy` turbulence model (it appears in two places in Make/files)
